js/control/ShareRoute.js

```diff
@@ -0,0 +1,149 @@
+BR.ShareRoute = L.Class.extend({
+    /**
+     * Sharing via Mastodon is currently disabled by default, because
+     * the share intent fails when the current route URL is longer
+     * than the post character limit for that instance.
+     */
+    options: {
+        services: {
+            mastodon: false,
+        },
+        shortcut: {
+            share_action: 65, // char code for 'a' ("action")
+        },
+    },
+
+    initialize: function () {
+        L.DomUtil.get('shareButton').onclick = L.bind(this.share, this);
+        L.DomEvent.addListener(document, 'keydown', this._keydownListener, this);
+    },
+
+    share: function (event) {
+        event.preventDefault();
+        this.services();
+        this.qrcode();
+    },
+
+    services: function () {
+        const self = this;
+
+        $('.share-copy-link').on('click', function () {
+            navigator.clipboard.writeText(self.getShareUrl());
+        });
+
+        $('a.share-service').each(function () {
+            $(this).attr('href', $(this).attr('href').replace('{url}', encodeURIComponent(self.getShareUrl())));
+        });
+
+        if (this.options.services.mastodon === true) {
+            $('.share-service-mastodon')
+                .removeAttr('hidden')
+                .on('click', function () {
+                    let mastodonServer = window.prompt(
+                        i18next.t('share.mastodon-enter-server-name'),
+                        'mastodon.social'
+                    );
+                    if (mastodonServer.indexOf('http') !== 0) {
+                        mastodonServer = 'https://' + mastodonServer;
+                    }
+                    window.open(mastodonServer + '/share?text=' + encodeURIComponent(self.getShareUrl()), '_blank');
+                });
+        }
+    },
+
+    /**
+     * Renders QR Code for the currect route:
+     *
+     * - hides the export modal dialog (QR Code is shown in a new dialog)
```

Comment:
doesn't apply to current implementation anymore?

Reply:
Yes, of course. I've updated this doc block.